code/go/internal/validator/semantic/validate_date.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"fmt"
+
+	"github.com/elastic/package-spec/v2/code/go/internal/errors"
+	"github.com/elastic/package-spec/v2/code/go/internal/fspath"
+)
+
+// ValidateDateFields verifies if date fields are of one of the expected types.
+func ValidateDateFields(fsys fspath.FS) errors.ValidationErrors {
+	return validateFields(fsys, validateDateField)
+}
+
+func validateDateField(fieldsFile string, f field) errors.ValidationErrors {
+	if f.Type != "date" && f.DateFormat != "" {
```

Comment:
@jillguyonnet @mrodm this looks like another case for the conditional json schemas :slightly_smiling_face: